src/services/videoconference/index.js

```diff
@@ -0,0 +1,399 @@
+ * @property {success:'SUCCESS'|'ERROR'} status indicator
+ * @property {[permissions:[String]]} user permissions
+ * @property {[error:String]} error message indication string
+ */
+
+class GetVideoconferenceServie extends VideoconferenceBaseService {
+	/**
+	 *
+	 * @param {String} id the id of a scope, the videoconference is related to
+	 * @param {Object} params
+	 * @param {String} params.route.scopeName the scope name for given scope id
+	 * @returns {VideoConference}
+	 */
+	async get(scopeId, params) {
+		const { scopeName } = params.route;
+
+		// PARAMETER VALIDATION
+		VideoconferenceBaseService.throwOnValidationErrors(scopeId, scopeName);
+
+		const { app } = this;
+		const authenticatedUser = await getUser({ params, app });
+
+		// CHECK PERMISSIONS
+		VideoconferenceBaseService.throwOnPermissionErrors(authenticatedUser);
+
+		// WIP BELOW
+
+		const responseTypes = ['wait', 'start', 'join', 'error'];
+		if (!params.query.demo || !responseTypes.includes(params.query.demo)) {
+			throw new BadRequest(`define demo in query with one value of${JSON.stringify(responseTypes)}`);
+		}
+
+		const createResponse = (status, state, permission, url, options) => ({
+			status, state, permission, url, options,
+		});
+
+		// sample wait  response
+		switch (params.query.demo) {
+			case 'wait':
+				return createResponse(
+					RESPONSE_STATUS.SUCCESS,
+					STATES.NOT_STARTED,
+					PERMISSIONS.JOIN_MEETING,
+				);
+			case 'start':
+				return createResponse(
+					RESPONSE_STATUS.SUCCESS,
+					STATES.READY,
+					PERMISSIONS.START_MEETING,
+					undefined,
+					{
+						moderatorMustApproveJoinRequests: false,
+						everybodyJoinsAsModerator: false,
+					},
+				);
+			case 'join':
+				return createResponse(
+					RESPONSE_STATUS.SUCCESS,
+					STATES.STARTED,
+					PERMISSIONS.JOIN_MEETING,
+					'https://some.url',
+				);
+			default: // error
+				return {
+					status: RESPONSE_STATUS.ERROR,
+					error: { message: 'errorMessage', key: 123 },
+				};
+		}
+	}
+}
+
+class CreateVideoconferenceService extends VideoconferenceBaseService {
+	/**
+ * creates an videoconference url to join a meeting, inside a scope defined by
+	 * id and scopeName depending on permissions as moderator or attendee.
+	 * this may fail due missing permissions
+	 * @param {{scopeName:string, id:string}} data
+	 * @param {Object} params
+	 * @param {Boolean} [params.moderatorMustApproveJoinRequests=false] - let moderators approve everybody who jons the videoconference
+	 * @param {Boolean} [params.everybodyJoinsAsModerator=false] - let everybody join the videoconference as moderator
+	 * @param {Boolean} [params.everyAttendeJoinsMuted=false] - let everybody except moderators join muted
+	 * @param {[String]} [params.rolesAllowedToAttendVideoconference=false] - scope roles who may attend the videoconference
+	 * @param {[String]} [params.rolesAllowedToStartVideoconference=false] - scope role who may start the videoconference
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [This line has a length of 121. Maximum allowed is 120.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4819756)